### ISSUE TITLE
Distance and Direction WAB widget: Ellipse tool does not create any ellipse graphics

### ISSUE THREAD

**Issue by ACueva** [Esri/solutions-webappbuilder-widgets#687]
_From @elinz on December 5, 2016 23:27_

Ran the ellipse tool from the Distance and Direction widget with the parameters shown in the screenshot below, and no ellipse was created. The only thing drawn on the map was the graphic to indicate the center point:
![image](https://cloud.githubusercontent.com/assets/4635490/20906871/2c502728-baff-11e6-8f6d-18f89b5553da.png)

## Steps to Reproduce (for bugs)

1. Open Distance and Direction widget and click on Ellipse tab
2. In "Center Point" parameter, enter 55.77 37.56 (which is Lat Long)
3. Set "Unit" parameter, select "Kilometers"
4. Set "Major Axis" to 1000
5. Set "Minor Axis" to 500
6. Set "Orientation Angle" to 45 and hit return while focus is in this textbox.

## Your Environment

* Version used: Web App Builder 2.2
* Environment name and version (e.g. Chrome 39, node.js 5.4):
* Operating System and version (desktop or mobile):
* Link to your project:

_Copied from original issue: Esri/military-tools-for-arcgis#75_

**Comment by kgonzago** [comment]
@elinz Can you provide which browser and version you were testing with?

**Comment by kgonzago** [comment]
@elinz Can you provide detailed steps to reproduce this issue?

**Comment by elinz** [comment]
@kgonzago Browser is Chrom 51.0.2704.79 m

**Comment by elinz** [comment]
@kgonzago Added steps above.

**Comment by elinz** [comment]
@kgonzago I've also tried this in IE 11.0.9600.18525 and it doesn't work

**Comment by kgonzago** [comment]
Thanks @elinz. I will look into it.

**Comment by adgiles** [comment]
@elinz @kgonzago I found you have to physically press enter on each of the input boxes in order. You should see the point drawn, then the major axis, minor axis then finally on the rotate the ellipse is drawn

**Comment by elinz** [comment]
@adgiles @kgonzago I've retested and found that you don't have to hit return on each of the input boxes. In my test...
1. Click in Center Point textbox. Entered coordinate. **I have to hit return**. (NOTE: the center point is never drawn as in my screenshot above).
2. Click in Major axis textbox. Enter value. Don't need to hit return.
3. Click in Minor axis textbox. Enter value. Don't need to hit return.
4. Click in Orientation Angle textbox. Enter value. **I have to hit return so ellipse is created**

The ellipse is created.

**Comment by adgiles** [comment]
@elinz @kgonzago 

Eric, correct the first time round you do not need to hit return on the Major and Minor text boxes as the line is created on the key up event (you will see it being drawn as you type in the number). If you want to create an additional ellipse at a different coordinate or angle with the same minor and major values you will have to click in the text box and hit return to fire the event which in turn draws the line.

**Comment by elinz** [comment]
@kgonzago @adgiles This tool is not working as I've described it or @adgiles has described. This is very painful to use.

I created a new app, added the widget, launched the app. Open the widget. Clicked on Ellipse tab and did the following...
1. Entered 55.7723N 037.5649E and hit return (circle is drawn at location).
2. Changed Unit to Kilometers.
3. Entered 1000 in Major axis and hit return (line is drawn as I'm typing in value).
4. Entered 500 in Minor axis and hit return (line is drawn as I'm typing in value).
5. Entered 45 in Orientation Angle and hit return (NO ellipse is created)

At this point I have to go back and hit return in the Center Point, Major, Minor, and Orientation Angle textboxes for the ellipse to be created.
